src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -584,6 +584,38 @@ namespace syrec {
 
         extend(tt);
 
+        // Refer to the one-pass synthesis  algorithm of https://www.cda.cit.tum.de/files/eda/2017_tcad_one_pass_synthesis_reversible_circuits.pdf.
+        if (onePass) {
+            if (m > n) {
+                // zeros are inserted to match the length of the output patterns.
+                augmentWithConstants(tt, m);
+            }
+
+            // k -> Minimum no. of additional lines required (no encoding is performed).
+            const auto [_, k] = encodeHuffman(tt, true);
+
+            const auto maxIO = std::max(n, m);
+
+            r           = (m + k) - maxIO;
+            totalNoBits = maxIO + r;
+
+            // based on the totalNoBits, zeros are appended to the inputs and the outputs.
+            augmentWithConstants(tt, totalNoBits, false, true, true);
+
+            ddSynth = std::make_unique<dd::Package<>>(totalNoBits);
+            qc      = std::make_shared<qc::QuantumComputation>(totalNoBits);
```

Comment:
no such call exists for the other branch of the `if`. why is that the case and is this really needed here?

Reply:
are you referring to the qc call? if so, I have modified it.